src/Emulator/Peripherals/Peripherals/Timers/ATCWDT200.cs

```diff
@@ -0,0 +1,410 @@
+//
+// Copyright (c) 2010-2023 Antmicro
+//
+// This file is licensed under the MIT License.
+// Full license text is available in 'licenses/MIT.txt'.
+//
+#define ATCWDT200_32BIT_TIMER
+using Antmicro.Renode.Core;
+using Antmicro.Renode.Peripherals.CPU;
+using Antmicro.Renode.Core.Structure.Registers;
+using Antmicro.Renode.Logging;
+using Antmicro.Renode.Time;
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Antmicro.Renode.Peripherals.Miscellaneous;
+using System.Threading;
+
+namespace Antmicro.Renode.Peripherals.Timers
+{
+  public class ATCWDT200 : BasicDoubleWordPeripheral, IKnownSize
+  {
+    public ATCWDT200(Machine machine) : base(machine)
+    {
+      DefineRegisters();
+
+      IRQ0 = new GPIO();
+      IRQ1 = new GPIO();
+
+
+      interruptTimer = new LimitTimer(machine.ClockSource, timerFrequency, this, "interrupt_timer", InitialLimit, workMode: WorkMode.OneShot, eventEnabled: true);
+      interruptTimer.LimitReached += () =>
+      {
+        this.InfoLog("interrupt limit reached");
+        interruptPending.Value = true;
+
+        UpdateInterrupts();
+        resetTimer.Enabled = true;
+        this.InfoLog("enable watchdog reset timer {0}", resetTimer.Enabled);
+
+      };
+
+      resetTimer = new LimitTimer(machine.ClockSource, timerFrequency, this, "reset_timer", InitialLimit, workMode: WorkMode.OneShot, eventEnabled: true);
+      resetTimer.LimitReached += () =>
+           {
+             IRQ1.Set();
+             this.InfoLog("Sending  reset signal {0}", IRQ1.IsSet);
+
+             this.InfoLog("reset limit reached");
+
+             // machine.RequestReset();
+
+           };
+
+    }
+
+    public override void Reset()
+    {
+      base.Reset();
+      interruptTimer.Reset();
+      resetTimer.Reset();
+      IRQ0.Unset();
+      IRQ1.Unset();
+      resetSequence = ResetSequence.WaitForUnlock;
+      this.InfoLog("Watchdog timer reset");
+
+    }
+
+
+    private bool CheckifUnlock(Registers reg)
+    {
+      if (registersUnlocked)
+      {
+        return true;
+
+      }
+
+      this.Log(LogLevel.Warning, "Writing to {0} is allowed only when the register is unlocked", reg);
+      return false;
+    }
+
+    private ulong resetinterval(int value)
```

Comment:
The naming of this function does not seem to follow the style from Renode. For instance, this one should be called ResetInterval